blog/_posts/2020-07-15-Tetiana-Dadakova.md

```diff
@@ -0,0 +1,232 @@
+---
+layout: post
+title: "Future blogpost about MABE"
+date: 2020-07-15
+author: Tetiana
+---
+
+# Genome change logging using range map  
+
+For more information about MABE (Modular Agent-Based Evolution platform) see [this introduction](https://szendejo.github.io/waves/blog/Team-MABE.html).
+
+## Genome Class
+
+### Naïve Implementation
+
+The naïve implementation of the genome is using a vector data structure from the standard library. The overwrite, insert and remove operations are, therefore, implemented using the standard library algorithms, specifically:
+```cpp
+std::vector<std::byte> sites{std::byte(1), std::byte(2), std::byte(3)}; // this genome consists of three sites with values 1, 2 and 3
+
+// Overwrite sites startin at index with values from segment
+virtual void overwrite(size_t index, const std::vector<std::byte>& segment) override {
+    for (size_t i(0); i < segment.size(); i++) {
+        sites[index + i] = segment[i];
+    }
+}
+
+// Insert mutation: a segment is inserted at index
+virtual void insert(size_t index, const std::vector<std::byte>& segment) override {
+    sites.insert(sites.begin() + index, segment.begin(), segment.end());
+}
+
+// Remove mutation: segmentSize sites are removed atarting at index 
+virtual void remove(size_t index, size_t segmentSize) override {
+    sites.erase(sites.begin() + index, sites.begin() + index + segmentSize);
+}
+
+```
+
+The advantages of such approach include:
+* All the sites are in contiguous memory -> fast operations due to cache-friendliness (e.g. random access or iterations)  
+* Use of C++ standard library data structures -> code is simple, readable, expressive and optimized for performance  
+  
+However, there are also disadvantages:  
+* Every generation, the whole genome (the whole `sites` vector) is copied and then the mutations are applied to it -> In a common situation of large genome and low mutation rates, it means copying a lot of values that didn't change  
+* The `insert()` and `erase()` algorithms have linear time complexity -> inefficient time  
+  
+  **The goal of this project was to investigate if storing only the mutations (as opposed to storing the whole genome) would provide a better time and memory performance.**
+
+### Optimized Implementation Using Change Log  
+
+One of the ways to improve the time complexity as well as optimize for memory use is to have a change log. The change log will keep track of the mutations that occurred between the parent and the offsprings over generations. This means only storing the differences between parent genome and it's offsprings as opposed to storing the whole genome for every offspring. 
```

Comment:
A small thing, but I think the proper markdown style is to have each sentence be on its own line for readability purposes